model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/const_holder_conv.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2022 Sony Semiconductors Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from torch.nn import Conv2d, ConvTranspose2d
+from torch.nn.functional import conv2d, conv_transpose2d
+from model_compression_toolkit.core.common.graph.graph_matchers import NodeOperationMatcher, EdgeMatcher
+from model_compression_toolkit.core import common
+from model_compression_toolkit.core.common.graph.base_graph import Graph
+from model_compression_toolkit.core.common.graph.base_node import BaseNode
+from model_compression_toolkit.core.pytorch.reader.graph_builders import ConstantHolder
+from model_compression_toolkit.core.pytorch.constants import IN_CHANNELS, OUT_CHANNELS, KERNEL_SIZE, KERNEL, BIAS, CONSTANT
+
+
+class ConstantHolderConv(common.BaseSubstitution):
+    """
+    Find "ConstantHolder" followed by conv layer to substitute a new single layer with weights if needed
+    """
+
+    def __init__(self):
+        """
+        Matches: 'ConstantHolder' followed by conv layer
+        """
+        first_node = NodeOperationMatcher(ConstantHolder)
+        second_node = NodeOperationMatcher(conv2d) | NodeOperationMatcher(conv_transpose2d)
+        super().__init__(matcher_instance=EdgeMatcher(first_node, second_node))
+
+    def substitute(self,
+                   graph: Graph,
+                   nodes: BaseNode) -> Graph:
+        """
+        Substitute ConstantHolder and conv/linear layer with single layer
+        Args:
+            graph: Graph we apply the substitution on.
+            nodes: nodes that match the pattern in the substitution init.
+
+        Returns:
+            Graph after applying the substitution.
+        """
+        first_node = nodes[0] # constant holder node
+        second_node = nodes[1] # convolution node
+
+        # Check if there is a connection
+        if graph.get_edge_data(first_node,second_node) is None:
+            return graph # skip substitution
+
+        # Set new layer
+        if second_node.type == conv2d:
+            NewLayer = Conv2d
+            out_channel_index = 1
+        elif second_node.type == conv_transpose2d:
+            NewLayer = ConvTranspose2d
+            out_channel_index = 0
```

Comment:
isn't the out_channel_index framework specific?

Reply:
I will use fw_info here